installation-bundle/src/Resources/config/services.yml

```diff
@@ -2,12 +2,35 @@ services:
     _defaults:
         autoconfigure: true
 
+    Contao\InstallationBundle\Database\Version447Update:
+        arguments:
+            - '@database_connection'
+            - '@translator'
+        tags:
+            - { name: contao.migration, priority: 99 }
+
+    Contao\InstallationBundle\Database\Version480Update:
+        arguments:
+            - '@database_connection'
+            - '%kernel.project_dir%'
+        tags:
+            - { name: contao.migration, priority: 99 }
+
+    Contao\InstallationBundle\Database\:
+        resource: '../../Database/Version*'
+        exclude: '../../Database/Version{480,447}Update.php'
```

Comment:
I guess we cannot use autowiring to make this a little less ugly?

Reply:
I don’t think so. `Version480Update` needs the `%kernel.project_dir%` which is just a string and AFAIK strings cannot be autowired?

Reply:
They can be autowired:
```yml
# services.yml
services:
    _defaults:
        bind:
            $projectDir: '%kernel.project_dir%'
```

Reply:
No autowiring in bundles ☝️ This here is "automatically loading classes" and not related to autowiring at all. So it's good as it is imho.

Reply:
I disagree. "No autowiring in bundles" in generally correct, however, this is a valid exception case IMHO, because loading version classes is a kind of closed sub-system within the application. And autowiring would be handy here.

But since this is legacy code that should be moved into the respective bundle anyway, I am fine with leaving it as is.

Reply:
Well...maybe we should distinguish here. I'm not saying autowiring is evil for all bundles. We just have to keep in mind that the `core-bundle` is being extended by third party developers and autowiring would just make things possibly inconsistent depending on the application setup.
However, in this case, we could extract the migrations into their own `migrations.yaml` file in which case they would not affect the rest of the services:

```yaml
services:
    _defaults:
        bind:
            string $projectDir: '%kernel.project_dir%'
    Contao\InstallationBundle\Database\:
        resource: '../../Database/Version*'
```

Reply:
Yes, that is what I had in mind. But as I said, it is fine for me the way it is as well.